### DISCUSSION TITLE
How to do column alias?

### DISCUSSION THREAD

**Discussion by dullbananas** [diesel-rs/diesel#3740] [Q&A]
Is there a way to do something like `EXISTS (...) AS thing_exists` in a selection so that `thing_exists` can be referenced in a filter?

**Comment by weiznich** [comment]
Column aliases are not supported in the built-in dsl yet, which means you have the following alternatives:

* Using `diesel::sql_query`
* Using `diesel::dsl::sql` for parts of your query
* Build your own dsl extension for this functionality.
